mlflow/types/llm.py

```diff
@@ -90,24 +127,73 @@ def from_dict(cls, data):
         return cls(**filtered_data)
 
 
+@dataclass
+class FunctionToolCallArguments(_BaseDataclass):
+    """
+    The arguments of a function tool call made by the model.
+
+    Args:
+        arguments (str): A JSON string of arguments that should be passed to the tool.
+        name (str): The name of the tool that is being called.
+    """
+
+    name: str
+    arguments: str
+
+    def __post_init__(self):
+        self._validate_field("name", str, True)
+        self._validate_field("arguments", str, True)
+
+    def to_tool_call(self, id):
+        return ToolCall(id=id, function=self)
+
+
+@dataclass
+class ToolCall(_BaseDataclass):
+    """
+    A tool call made by the model.
+
+    Args:
+        id (str): The ID of the tool call.
+        function (:py:class:`FunctionToolCallArguments`): The arguments of the function tool call.
+        type (str): The type of the object. Currently only "function" is supported.
+    """
+
+    id: str
```

Comment:
Wondering if we can generate a default value here? Otherwise `to_tool_call` still need to pass the id 🤔

Reply:
yeah i think we can do that—it seems the actual ID is not so important here, the client just needs a way to reference the tool call message. i'll add it!